File: qe-selenium-python/Activity_Fourteen.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains

driver = webdriver.Firefox()

driver.get("https://training-support.net/webelements/tables")
driver.maximize_window()

print(driver.title)

print("Number of rows:", len(driver.find_elements(By.XPATH, "//table//tr")))
print("Number of columns:", len(driver.find_elements(By.XPATH, "//table//tr[1]/td")))

before = driver.find_element(By.XPATH, "//table//tr[5]//td[2]").text
print("Book Name in Fifth Row:", before)

actions = ActionChains(driver)
actions.click(driver.find_element(By.XPATH, "//table//thead//tr[1]//th[5]")).perform()

after = driver.find_element(By.XPATH, "//table//tr[5]//td[2]").text
print("Book Name in Fifth Row:", after)

driver.quit()
    

File: qe-selenium-java/src/main/java/demos/ActivityFourteen.java
package demos;

import java.util.*;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class ActivityFourteen {

    public static void main(String[] args) throws InterruptedException {
        WebDriver driver = new FirefoxDriver();

        driver.get("https://training-support.net/webelements/tables");
        driver.manage().window().maximize();

        System.out.println(driver.getTitle());

        System.out.println("Number of rows: " + driver.findElements(By.xpath("//table//tr")).size());

        System.out.println("Number of columns: " + driver.findElements(By.xpath("//table//tr[1]/td")).size());

        System.out
                .println("Book Name in Fifth Row: " + driver.findElement(By.xpath("//table//tr[5]//td[2]")).getText());

        Actions actions = new Actions(driver);

        actions.click(driver.findElement(By.xpath("//table//thead//tr[1]//th[5]"))).perform();

        System.out
                .println("Book Name in Fifth Row: " + driver.findElement(By.xpath("//table//tr[5]//td[2]")).getText());

        driver.quit();
    }

}
